jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamDebugSupport.kt

```diff
@@ -45,7 +56,35 @@ interface SamDebugSupport {
         state: SamRunningState,
         debugHost: String,
         debugPorts: List<Int>
-    ): Promise<XDebugProcessStarter?> = resolvedPromise(createDebugProcess(environment, state, debugHost, debugPorts))
+    ): Promise<XDebugProcessStarter?> {
+        val promise = AsyncPromise<XDebugProcessStarter?>()
+        val bgContext = ExpirableExecutor.on(AppExecutorUtil.getAppExecutorService()).expireWith(environment).coroutineDispatchingContext()
+
+        ApplicationThreadPoolScope(environment.runProfile.name).launch(bgContext) {
+            try {
+                val debug = createDebugProcess(environment, state, debugHost, debugPorts)
+
+                runInEdt {
+                    promise.setResult(debug)
```

Comment:
any reason why this needs to happen on EDT?

Reply:
Yes, it will throw if it is not set on EDT (this is JB's special promise implementation)